enhancements/network/external-lb-vips.md

```diff
@@ -0,0 +1,268 @@
+   will set the fields for `apiVIP` and `ingressVIP`. Since this is will be TechPreview at first,
+   customers will have to set `featureSet: TechPreviewNoUpgrade`.
+3. In the installer, if the platform does support external loadbalancer, some validation will be run
+   to check that the VIPs can already be resolved (the validation already exists for vsphere, we need to
+    make it re-usable). The new field for `LoadBalancerType` will be set in `PlatformStatus`.
+4. In machine-config-operator, if LB type is external, the HAproxy and Keepalived will not be
+   deployed.
+
+### API Extensions
+
+A new field will be added to the `PlatformStatus` of the infrastructure object to tell
+the kind of the loadbalancer.
+
+```golang
+// PlatformLoadBalancerType defines the type of load balancer used by the cluster.
+type PlatformLoadBalancerType string
+const (
+    // LoadBalancerTypeUserManaged is a loadbalancer that is managed outside of the cluster by the customer.
+    LoadBalancerTypeUserManaged PlatformLoadBalancerType = "UserManaged"
+
+    // LoadBalancerTypeOpenShiftManaged is a loadbalancer that is managed by the OpenShift cluster.
+    LoadBalancerTypeOpenShiftManaged PlatformLoadBalancerType = "OpenShiftManaged"
+)
+
+type VSpherePlatformStatus struct {
+    // loadBalancerType defines if a user-managed or openshift-managed loadbalancer
+    // is to be used for the OpenShift API and Ingress endpoints.
+    // When set to OpenShiftManaged the static pods in charge of API and Ingress traffic load-balancing
+    // defined in the machine config operator will be deployed.
+    // When set to UserManaged these static pods will not be deployed and it is expected that
+    // the load balancer is configured out of band by the deployer.
+    // When omitted, this means no opinion and the platform is left to choose a reasonable default.
+    // The default value is OpenShiftManaged.
+    // +default="OpenShiftManaged"
+    // +kubebuilder:default:="OpenShiftManaged"
+    // +kubebuilder:validation:Enum:="OpenShiftManaged";"UserManaged"
+    // +kubebuilder:validation:XValidation:rule="oldSelf == '' || self == oldSelf",message="loadBalancerType is immutable once set"
+    // +openshift:enable:FeatureSets=TechPreviewNoUpgrade
+    // +optional
+    LoadBalancerType PlatformLoadBalancerType `json:"loadBalancerType"`
+}
+```
+
+We decided to put this field in `PlatformStatus`. This is because the `Platform` field will probably have to be validated, which
+is common for the infrastructure object. For example, to validate that a transition is valid, or does the cluster accept that change.
+We will need the field consumers to consume from `PlatformStatus` rather than `PlatformSpec`.
+This will allow the user to express their desire to change the `PlatformSpec`, but the cluster could then reject it thanks to the validation.
```

Comment:
This is normally the preserve of a validating webhook.

Reply:
In the infrastructure object, its done by a number of controllers already (for each respective config that needs validating). There are some cases where validating webhooks and synchronous validation doesn't work well. Having status also allows you to assert that the feature has migrated, so we will need something there anyway, a platform accepted value, and then some way to report that individual operators have acted upon this value